asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py

```diff
@@ -0,0 +1,778 @@
+        if not loop_db.exists():
+            raise ValueError(f"Loop database file does not exist: {args.loop_db}")
+
+    if args.seqres_yaml is not None:
+        seqres_yaml = Path(args.seqres_yaml)
+        # check it exists
+        if not seqres_yaml.exists():
+            raise ValueError(f"SEQRES yaml file does not exist: {args.seqres_yaml}")
+
+        # load it
+        logger.info(f"Using SEQRES from {args.seqres_yaml}")
+        with open(args.seqres_yaml) as f:
+            seqres_dict = yaml.safe_load(f)
+        seqres = seqres_dict["SEQRES"]
+    else:
+        seqres = None
+
+    # load the receptor
+    receptor_name = receptor.stem
+    xtal = CrystalCompoundData(
+        str_fn=args.receptor, smiles=None, output_name=str(receptor_name)
+    )
+
+    logger.info(f"Loaded receptor {receptor_name} from {receptor}")
+    oe_prep_function(
+        xtal, args.ref_prot, seqres, prep_dir, args.loop_db, args.protein_only
+    )
+    logger.info(f"Finished prepping receptor at {datetime.now().isoformat()}")
+
+    # grab the files that were created
+    prepped_oedu = (
+        prep_dir / f"{receptor_name}" / f"{receptor_name}_prepped_receptor_0.oedu"
+    )
+    prepped_pdb = (
+        prep_dir / f"{receptor_name}" / f"{receptor_name}_prepped_receptor_0.pdb"
+    )
+
+    # check the prepped receptors exist
+    if not prepped_oedu.exists() or not prepped_pdb.exists():
+        raise ValueError(
+            f"Prepped receptor files do not exist: {prepped_oedu}, {prepped_pdb}"
+        )
+
+    logger.info(f"Prepped receptor: {prepped_pdb}, {prepped_oedu}")
+
+    # read design unit and split it
+    du = oechem.OEDesignUnit()
+    oechem.OEReadDesignUnit(str(prepped_oedu), du)
+
+    # extract the ligand, protein and complex
+    lig, protein, lig_prot_complex = split_openeye_design_unit(du)
+
+    # write out the protein
+    protein_path = prep_dir / f"{receptor_name}" / f"{receptor_name}_protein.pdb"
+    logger.info(f"Writing out protein to {protein_path}")
+    save_openeye_pdb(protein, str(protein_path))
+
+    # write out the complex
+    complex_path = prep_dir / f"{receptor_name}" / f"{receptor_name}_complex.pdb"
+    logger.info(f"Writing out complex to {complex_path}")
+    save_openeye_pdb(lig_prot_complex, str(complex_path))
+
+    # write out the ligand
+    ligand_path = prep_dir / f"{receptor_name}" / f"{receptor_name}_ligand.sdf"
+    logger.info(f"Writing out ligand to {ligand_path}")
+    save_openeye_sdf(lig, str(ligand_path))
+
+    ligand_smiles = oechem.OEMolToSmiles(lig)
+
+    logger.info(f"Xtal ligand: {ligand_smiles}")
+
+    # setup docking
+    logger.info(f"Starting docking setup at {datetime.now().isoformat()}")
+
+    dock_dir = output_dir / "docking"
+    dock_dir.mkdir(parents=True, exist_ok=True)
+    intermediate_files.append(dock_dir)
+
+    # ML stuff for docking, fill out others as we make them
+    logger.info("Setup ML for docking")
+    gat_model_string = "asapdiscovery-GAT-2023.04.12"
```

Comment:
Can you add some visual indicators for where each of the steps (input process, target prep, docking, artifact generation) happen in the script? Something like

```python

#### docking ##########

```